test/functional/bulk_tests.js

```diff
@@ -71,6 +71,64 @@ describe('Bulk', function() {
     }
   });
 
+  it('should ignore undefined values in unordered bulk operation if `ignoreUndefined` specified', {
+    metadata: {
+      requires: { topology: ['single'] }
+    },
+
+    test: function() {
+      const client = this.configuration.newClient(this.configuration.writeConcernMax(), {
+        poolSize: 1
+      });
+
+      return client
+        .connect()
+        .then(client => {
+          const db = client.db(this.configuration.db);
+          const col = db.collection('batch_write_unordered_ops_1');
+
+          return col
+            .initializeUnorderedBulkOp({ ignoreUndefined: true })
+            .insert({ a: 1, b: undefined })
+            .execute()
+            .then(() => col.find({}).toArray())
+            .then(docs => {
+              expect(docs[0]['a']).to.equal(1);
+              expect(docs[0]['b']).to.not.exist;
+            });
+        })
+        .then(() => client.close());
+    }
+  });
+
+  it('should ignore undefined values in ordered bulk operation if `ignoreUndefined` specified', {
+    metadata: {
+      requires: { topology: ['single'] }
+    },
+
+    test: function() {
+      var client = this.configuration.newClient(this.configuration.writeConcernMax(), {
+        poolSize: 1
+      });
+
+      return client.connect().then(client => {
+        var db = client.db(this.configuration.db);
+        var col = db.collection('batch_write_ordered_ops_1');
+
+        return col
+          .initializeOrderedBulkOp({ ignoreUndefined: true })
+          .insert({ a: 1, b: undefined })
+          .execute()
+          .then(() => col.find({}).toArray())
```

Comment:
nitpick: we can use `findOne` instead of `find({}).toArray()` to return a single document - I don't think we need to change this, but just consider it in the future

Reply:
Yeah 👍